### PR TITLE
🌱 Add dependabot configuration

### PR DESCRIPTION
**Pull request by kashifest**
Add dependabot configuration

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "weekly"
+      commit-message:
+      prefix: ":seedling:"
```
Comment (lentzi90) [inline .github/dependabot.yml:11]: I think prefix should be indented one more level (same as line 28).

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  # together with controller-runtime.
+  - dependency-name: "k8s.io/*"
+  - dependency-name: "go.etcd.io/*"
+  - dependency-name: "google.golang.org/grpc"
+  commit-message:
+    prefix: ":seedling:"
+  labels:
+    - "area/dependency"
+    - "ok-to-test"
+
```
Comment (lentzi90) [inline .github/dependabot.yml:32]: Looks like extra whitespace here that makes github complain about no newline at the end of the file

File: .github/dependabot.yml (hunk 1 above)
Comment (kashifest) [reply .github/dependabot.yml]: thanks, actually the prvious line was wrongly indented

File: .github/dependabot.yml (hunk 2 above)
Comment (kashifest) [reply .github/dependabot.yml]: done

**Review by lentzi90 — COMMENTED** [review]
/approve

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  - dependency-name: "sigs.k8s.io/controller-runtime"
+  # Ignore k8s and its transitives modules as they are upgraded manually
+  # together with controller-runtime.
+  - dependency-name: "k8s.io/*"
+  - dependency-name: "go.etcd.io/*"
+  - dependency-name: "google.golang.org/grpc"
+  commit-message:
+    prefix: ":seedling:"
+  labels:
+    - "area/dependency"
```
Comment (tuminoid) [inline .github/dependabot.yml:30]: do we have this label?

**Review by Sunnatillo — COMMENTED** [review]
LGTM

File: .github/dependabot.yml (hunk 3 above)
Comment (kashifest) [reply .github/dependabot.yml]: good point, thought of adding it first , but it has to be in prow, dropping it for now.

**Review by tuminoid — APPROVED** [review]
/lgtm

**Pull request merged**